feature_engine/creation/math_features.py

```diff
@@ -202,14 +265,117 @@ def transform(self, X: pd.DataFrame) -> pd.DataFrame:
         X_new: Pandas dataframe, shape = [n_samples, n_features + n_operations]
             The input dataframe plus the new variables.
         """
+
+        def np_transform(np_df, new_variable_names, np_variables, np_functions):
```

Comment:
we need typehint and better function names. What is np_transform supposed to do? not immediately obvious from the name. I'd also not define a method within a method. Let's define it outside preceded by _